### PR TITLE
treewide: Remove Consul support

### PR DESCRIPTION
**Pull request by twpayne**
As announced in the 1.11 release, remove support for the Consul
key-value store.

This commit removes Consul support only. Removing the pkg/kvstore
abstraction would be for a follow-up PR, after discussion.

Signed-off-by: Tom Payne <tom@isovalent.com>

### CODE REVIEW

**Comment by twpayne** [comment]
Note that the build-commits /  Check step currently fails as this PR is split into two commits to make rebasing easier. Before merging, they can be squashed together.

File: pkg/kvstore/etcd_test.go [hunk 1]
```diff
@@ -199,7 +199,7 @@ endpoints:
 	}{
 		{
 			args: args{
-				backend: consulName,
+				backend: "not-etcd",
```
Comment (michi-covalent) [inline pkg/kvstore/etcd_test.go:202]: :joy_cat:

**Review by michi-covalent — COMMENTED** [review]
:heavy_check_mark: the change itself looks good to me, but i'd like to discuss https://github.com/cilium/cilium/issues/18851 in the community meeting before we go ahead and remove the consul support.

**Comment by twpayne** [comment]
As just agreed in the community meeting, we'll maintain Consul support for 1.12 and review again for 1.13. So, closing this PR for now.

**Pull request closed without merging**